com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ServiceProgressReporter.java

```diff
@@ -0,0 +1,63 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import static java.text.MessageFormat.format;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperation;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.core.cf.services.TypedServiceOperationState;
+import com.sap.cloud.lm.sl.cf.process.steps.ExecutionWrapper;
+
+@Component
+public class ServiceProgressReporter {
+
+    public void reportOverallProgress(ExecutionWrapper execution, List<ServiceOperation> lastServicesOperationS,
+                                      Map<String, ServiceOperationType> triggeredServiceOperations) {
+        List<TypedServiceOperationState> nonFinalStates = getNonFinalStates(lastServicesOperationS);
+        List<String> nonFinalStateStrings = getStateStrings(nonFinalStates);
+
+        int doneOperations = triggeredServiceOperations.size() - nonFinalStates.size();
+        if (!nonFinalStateStrings.isEmpty()) {
+            execution.getStepLogger()
+                     .info("{0} of {1} done, ({2})", doneOperations, triggeredServiceOperations.size(),
+                           String.join(",", nonFinalStateStrings));
+        } else {
+            execution.getStepLogger()
+                     .info("{0} of {0} done", triggeredServiceOperations.size());
```

Comment:
Why this and the above log messages are different?

Reply:
Because variables are different for both messages